stan/math/rev/core/autodiffstackstorage.hpp

```diff
@@ -61,26 +83,25 @@ struct AutodiffStackSingleton {
   explicit AutodiffStackSingleton(AutodiffStackSingleton_t const &) = delete;
   AutodiffStackSingleton &operator=(const AutodiffStackSingleton_t &) = delete;
 
-  static inline AutodiffStackStorage &instance() {
-#ifdef STAN_THREADS
-    thread_local static AutodiffStackStorage instance_;
-#endif
-    return instance_;
+  constexpr static inline AutodiffStackStorage &instance() {
+    return *instance_;
   }
 
-#ifndef STAN_THREADS
-
- private:
```

Comment:
Shoudl this pointer still be private?

Reply:
Hmpf... in principle it can be private right now, yes. In my follow-up on this I do need to access it. We seem to be relatively liberal with access to class members which is why I took out the private. I think it also makes testing easier.

I would opt to leave it as is for developer convenience. In case you think it is important to make it private we can probably do that. How about a comment in the code saying that one should not mess with this?